server/channels/app/export.go

```diff
@@ -131,27 +131,35 @@ func (a *App) BulkExport(ctx request.CTX, writer io.Writer, outPath string, job
 	}
 
 	if opts.IncludeAttachments {
+		totalExportedFiles := 0
+		totalFiles := len(attachments) + len(directAttachments) + len(emojiPaths)
+
 		ctx.Logger().Info("Bulk export: exporting file attachments")
-		for _, attachment := range attachments {
-			if err := a.exportFile(outPath, *attachment.Path, zipWr); err != nil {
-				return err
-			}
+		if err = a.exportAttachments(ctx, attachments, outPath, zipWr, &totalExportedFiles, totalFiles); err != nil {
+			return err
 		}
-		for _, attachment := range directAttachments {
-			if err := a.exportFile(outPath, *attachment.Path, zipWr); err != nil {
-				return err
-			}
+
+		ctx.Logger().Info("Bulk export: exporting direct file attachments")
+		if err = a.exportAttachments(ctx, directAttachments, outPath, zipWr, &totalExportedFiles, totalFiles); err != nil {
+			return err
 		}
+
+		ctx.Logger().Info("Bulk export: exporting custom emojis")
 		for _, emojiPath := range emojiPaths {
 			if err := a.exportFile(outPath, emojiPath, zipWr); err != nil {
 				return err
 			}
+			totalExportedFiles++
+			if totalExportedFiles%10 == 0 {
+				ctx.Logger().Info("Bulk export: exporting file attachments progress", mlog.Int("total_exported_attachments", totalExportedFiles), mlog.Int("total_attachments", totalFiles))
```

Comment:
Should the log message make it clear that it's about emojis?

Reply:
I think I might change the log message say `files` instead of `attachments` so that it's more clear. See https://github.com/mattermost/mattermost/pull/26396/files#r1513470592

Reply:
The log message change makes sense. :+1: 

Can you please clarify what exactly gets exported here? Is it the custom emoji files?

Reply:
Yes it is the custom emoji files that are exported in this segment

Reply:
Changed the log message to mention emojis instead of files